std/os/index.zig

```diff
@@ -864,6 +875,10 @@ pub fn getCwd(out_buffer: *[MAX_PATH_BYTES]u8) GetCwdError![]u8 {
             const err = posix.getErrno(posix.getcwd(out_buffer, out_buffer.len));
             switch (err) {
                 0 => return cstr.toSlice(out_buffer),
+                posix.EACCES => return error.AccessDenied,
+                posix.EINVAL => unreachable,
+                posix.ENOENT => return error.FileNotFound,
+                posix.ENOMEM => return error.SystemResources,
```

Comment:
I believe this zig code is correct, but, imagine asking the kernel for the current working directory, and the response is "sorry, I can't tell you, I'm out of memory". Seems like an unfortunate limitation in the kernel design to me :-(

Reply:
This behaviour is shared by NetBSD and FreeBSD, according to their man pages.